invenio_rdm_records/contrib/journal/custom_fields.py

```diff
@@ -12,14 +12,199 @@
+        """Adds serialized journal data to the input data under the 'sources' key."""
+        _original = original or {}
+        custom_fields = _original.get("custom_fields", {})
+        journal_data = custom_fields.get("journal:journal", {})
+        title = journal_data.get("title")
+        volume = journal_data.get("volume")
+        issue = journal_data.get("issue")
+        pages = journal_data.get("pages")
+        year = _original.get("metadata", {}).get("publication_date")
+
+        if volume and issue:
+            volume = f"{volume}({issue})"
+
+        if not volume:
+            volume = issue
+
+        parts = [
+            title,
+            volume,
+            pages,
+            f"({year})" if year else None,
+        ]
+
+        # Update input data with serialized data
+        serialized_data = ", ".join([x for x in parts if x])
+        sources = data.get("sources", [])
+        sources.append(serialized_data)
+        data["sources"] = sources
+
+        return data
+
+
+class JournalMarcXMLDumper(FieldDumper):
+    """Dump processor for MarcXML serialization of 'Journal' custom field."""
+
+    def post_dump(self, data, original=None, **kwargs):
+        """Adds serialized journal data to the input data under the '773' key."""
+        _original = original or {}
+        custom_fields = _original.get("custom_fields", {})
+        journal_data = custom_fields.get("journal:journal", {})
+
+        if not journal_data:
+            return data
+
+        items_dict = {}
+
+        title = journal_data.get("title")
+        if title:
+            items_dict["p"] = title
+
+        volume = journal_data.get("volume")
+        if volume:
+            items_dict["v"] = volume
+
+        issue = journal_data.get("issue")
+        if issue:
+            items_dict["n"] = issue
+
+        pages = journal_data.get("pages")
+        if pages:
+            items_dict["c"] = pages
+
+        year = _original.get("metadata", {}).get("publication_date")
+        if year:
+            items_dict["y"] = year
+
+        if not items_dict:
+            return data
+
+        # TODO in zenodo, journal field serializes to
+        # TODO code 909, C, 4 but that's not in the specification
+        code = "773  "
+
+        data[code] = items_dict
+        return data
+
+
+class JournalCSLDumper(FieldDumper):
+    """Dump processor for CSL serialization of 'Journal' custom field."""
+    def post_dump(self, data, original=None, **kwargs):
+        """Adds serialized journal data to the input data."""
+        _original = original or {}
+        custom_fields = _original.get("custom_fields", {})
+        journal_data = custom_fields.get("journal:journal", {})
+        title = journal_data.get("title")
+        volume = journal_data.get("volume")
+        issue = journal_data.get("issue")
+        pages = journal_data.get("pages")
+
+        if not journal_data:
+            return data
+
+        data["container_title"] = title
```

Comment:
what if they are `None`. Do we want those fields there? or only those with values?